query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala

```diff
@@ -49,6 +49,71 @@ object RateFunctions {
       Some(result)
     }
   }
+
+  def derivFunction(window: Window): Option[Double] = {
```

Comment:
Avoid Option

Reply:
What should be used instead here?

Reply:
The caller does 
```
result.getOrElse(Double.NaN)
```
the NaN needs to be brought up here. Yes, a bit ugly but better to avoid the Option.

Reply:
BTW, I should not have simply said "avoid Option".
I actually like the scala Option instead of passing around nulls (or other values). But using Option here is a hot code path is what needs to be reconsidered.

Reply:
👍  will change it